### PR TITLE
[5G: MVC][AMF][SessionD] Sessiond-AMF Client

### PR DESCRIPTION
**Pull request by ronit-kumar-acl**
## Summary

Client to connect to AMF using amf_service as the channel name.
Requires PR: https://github.com/magma/magma/pull/3121 and https://github.com/magma/magma/pull/3123

## Test Plan

Unit Test logs: [amf_server_unit_test.log](https://github.com/magma/magma/files/5383827/amf_server_unit_test.log)

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Comment by themarwhal** [comment]
Please run `make precommit_sm` to run format+tests.

**Comment by themarwhal** [comment]
Oh, my bad it is OAI code. Then I think the command is `make format_oai`

**Review by themarwhal — APPROVED** [review]
SMF change looks good to me

**Review by uri200 — APPROVED** [review]

File: lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.h [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#include <grpc++/grpc++.h>
+#include <grpcpp/impl/codegen/status.h>
+
+#include "lte/protos/session_manager.grpc.pb.h"
+#include "lte/protos/spgw_service.grpc.pb.h"
+#include "lte/protos/policydb.pb.h"
+
+extern "C" {
+#include "spgw_service_handler.h"
+#include "log.h"
```
Comment (ssanadhya) [inline lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.h:23]: We don't need these two headers here, they are only relevant in the .cpp source.

File: lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.h [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ }  // namespace lte
+}  // namespace magma
+
+using grpc::ServerContext;
+using magma::lte::SetSMSessionContextAccess;
+using magma::lte::SmContextVoid;
+using magma::lte::SmfPduSessionSmContext;
+
+namespace magma {
+using namespace lte;
```
Comment (ssanadhya) [inline lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.h]:
```suggestion
namespace lte {
```

File: lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+namespace magma {
+using namespace lte;
+
+AmfServiceImpl::AmfServiceImpl() {}
+//Set message from SessionD received
+Status AmfServiceImpl::SetSmfSessionContext(
+    ServerContext* context, const SetSMSessionContextAccess* request,
+    SmContextVoid* response) {
+  OAILOG_INFO(LOG_UTIL, "Received  GRPC SetSMSessionContextAccess request\n");
```
Comment (ssanadhya) [inline lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp]:
```suggestion
  OAILOG_INFO(LOG_UTIL, "Received GRPC SetSMSessionContextAccess request\n"); 
```

File: lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+*/
+
+#include <string>
+
+#include "lte/protos/session_manager.pb.h"
+#include "lte/protos/spgw_service.pb.h"
+#include <folly/IPAddress.h>
+
+extern "C" {
+#include "spgw_service_handler.h"
```
Comment (ssanadhya) [inline lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp:19]: Why is this header needed here? If it is needed for future code, please include it in future PRs.

File: lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <string>
+
+#include "lte/protos/session_manager.pb.h"
+#include "lte/protos/spgw_service.pb.h"
```
Comment (ssanadhya) [inline lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp:15]: Why is this header needed here? If it is needed for future code, please include it in future PRs.

**Review by ssanadhya — CHANGES_REQUESTED** [review]

File: lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.h (hunk 1 above)
Comment (ronit-kumar-acl) [reply lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.h:23]: Sure, I'll test and push with changes.

File: lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp (hunk 4 above)
Comment (ronit-kumar-acl) [reply lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp:19]: yeah, this is not required, I'll remove and push the changes.

File: lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp (hunk 5 above)
Comment (ronit-kumar-acl) [reply lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp:15]: Yeah, this is not required, I'll remove and push the changes.

**Comment by gurrapualt** [comment]
This code pulled as part of #3366 request, so closing this.

**Pull request closed without merging**